shared/components/ember-output.jsx

```diff
@@ -0,0 +1,82 @@
+/* global React */
+
+export default React.createClass({
+    displayName: 'EmberOutput',
+
+    propTypes: {
+        source  : React.PropTypes.string.isRequired,
+        context : React.PropTypes.object.isRequired
+    },
+
+    injectMessages: function (locale, messages) {
+      var container  = this.container;
+      var lookupName = locale.toLowerCase();
+
+      if (container.has('locale:' + lookupName)) {
+          container.unregister('locale:' + lookupName);
+      }
+
+      container.register('locale:' + lookupName, {
+          locale:   locale,
+          messages: messages
+      }, {
+          instantiate: false
+      });
+    },
+
+    componentWillReceiveProps: function (nextProps) {
+        if (nextProps.source !== this.props.source) {
+            this.setState({});
+        }
+
+        var locale = Ember.get(nextProps, 'locales');
+
+        if (typeof locale === 'string') {
+            this.injectMessages(locale, nextProps.messages);
+            this.app.intl.set('locales', Ember.makeArray(locale));
+        }
+
+        this.controller.setProperties(nextProps);
+    },
+
+    componentDidMount: function () {
```

Comment:
@ericf do I need to do some cleanup in `componentWillUnmount` - like destroy the application?